Modules/_ctypes/callbacks.c

```diff
@@ -146,47 +146,55 @@ static void _CallPythonObject(void *mem,
                               int flags,
                               void **pArgs)
 {
-    Py_ssize_t i;
-    PyObject *result;
-    PyObject *arglist = NULL;
-    Py_ssize_t nArgs;
+    PyObject *result = NULL;
+    PyObject **args = NULL;
+    Py_ssize_t i = 0, j = 0, n_args = 0;
     PyObject *error_object = NULL;
     int *space;
     PyGILState_STATE state = PyGILState_Ensure();
 
-    nArgs = PySequence_Length(converters);
+    n_args = PySequence_Length(converters);
     /* Hm. What to return in case of error?
        For COM, 0xFFFFFFFF seems better than 0.
     */
-    if (nArgs < 0) {
+    if (n_args < 0) {
         PrintError("BUG: PySequence_Length");
         goto Done;
     }
 
-    arglist = PyTuple_New(nArgs);
-    if (!arglist) {
-        PrintError("PyTuple_New()");
-        goto Done;
+    PyObject *args_stack[_PY_FASTCALL_SMALL_STACK];
+    if (n_args <= (Py_ssize_t)Py_ARRAY_LENGTH(args_stack)) {
+        args = args_stack;
     }
-    for (i = 0; i < nArgs; ++i) {
-        /* Note: new reference! */
-        PyObject *cnv = PySequence_GetItem(converters, i);
+    else {
+        args = PyMem_Malloc(n_args * sizeof(PyObject *));
+        if (args == NULL) {
+            PyErr_NoMemory();
+            goto Done;
+        }
+    }
+
+    PyObject **cnvs = PySequence_Fast_ITEMS(converters);
+    for (i = 0; i < n_args; i++) {
+        PyObject *cnv = cnvs[i];
         StgDictObject *dict;
-        if (cnv)
-            dict = PyType_stgdict(cnv);
-        else {
+
+        if (cnv == NULL) {
             PrintError("Getting argument converter %zd\n", i);
             goto Done;
         }
```

Comment:
If converters is a tuple, cnv cannot be null. You can remove it. Maybe add a test at the function entry, to check that n_args <= the tuple length. But I expect this test to be done already in the caller functions, no?